babel/dates.py

```diff
@@ -960,9 +966,14 @@ def _iter_patterns(a_unit):
             # This really should not happen
             if pattern is None:
                 return ''
-            return pattern.replace('{0}', str(value))
-
-    return ''
+            if (depth=='shallow'):
+                formatted_string = ' '.join(filter(None, [formatted_string, pattern.replace('{0}', str(value))]))
```

Comment:
I'm not sure I understand what's going on in this magic incantation 😄 

It's repeated two lines underneath this, so maybe it should be DRYed out too?